src/services/utilities.ts

```diff
@@ -1332,19 +1332,20 @@ namespace ts {
             return n;
         }
 
-        const candidate = findRightmostChildNodeWithTokens(children, /*exclusiveStartPosition*/ children.length, sourceFile);
+        const candidate = findRightmostChildNodeWithTokens(n, /*exclusiveStartPosition*/ children.length, sourceFile);
         return candidate && findRightmostToken(candidate, sourceFile);
     }
 
     /**
-     * Finds the rightmost child to the left of `children[exclusiveStartPosition]` which is a non-all-whitespace token or has constituent tokens.
+     * Finds the rightmost child to the left of `n.children[exclusiveStartPosition]` which is a non-all-whitespace token or has constituent tokens.
      */
-    function findRightmostChildNodeWithTokens(children: Node[], exclusiveStartPosition: number, sourceFile: SourceFile): Node | undefined {
+    function findRightmostChildNodeWithTokens(n: Node, exclusiveStartPosition: number, sourceFile: SourceFile): Node | undefined {
+        const children = n.getChildren(sourceFile);

```

Comment:
Not sure if I should bother leaving `children` as a parameter to save a little bit of processing; seemed more clear to me to just pass down the parent node and get the children, rather than passing the parent with its children (all call sites technically have the list).

Reply:
It looks like `n` is only needed for its kind and that kind is only needed for an assert so, personally, I'd add a new parameter for the kind to make it clear that it's "extra".